run_project_tests.py

```diff
@@ -106,10 +154,12 @@ def display_name(self) -> str:
             return '{}   ({})'.format(self.path.as_posix(), self.name)
         return self.path.as_posix()
 
-    def __lt__(self, other: T.Any) -> T.Union[bool, type(NotImplemented)]:
+    def __lt__(self, other: T.Any) -> bool:
         if isinstance(other, TestDef):
             # None is not sortable, so replace it with an empty string
-            return (self.path, self.name or '') < (other.path, other.name or '')
+            s_id = int(self.path.name.split(' ')[0])
+            o_id = int(other.path.name.split(' ')[0])
+            return (s_id, self.path, self.name or '') < (o_id, other.path, other.name or '')
```

Comment:
do we really need to compare path if we're already comparing the number in the path?

Reply:
We already had cases that the same number was used twice after merging two PR's, and the name will be `Null` for non test matrix tests.

Reply:
Ah, okay that makes sense.